### PR TITLE
Redesign Video List as Table

### PR DESCRIPTION
**Pull request by aza547**
Still need to...
- [x] Tidy up all the table components (I did the raid one already)
- [x] Add indicators for sorting directions
- [x] Replace greyed out disk / cloud buttons with corresponding upload/download buttons.
- [x] Consider making row expansion and selection the same concept. 
- [x] Test deleting / undefined behaviour.
- [x] Fix persistent progress carried between povs
- [x] Make viewpoints field show first player? and "+N more"?
- [x] Resolve inline comments on this PR.
- [x] Make double clicking on a row expand/collapse it.
- [x] Expanded state needs incremented on a new video showing up?
- [x] Switch over to using unique ID to hold playing video instead of the mess currently.

### CODE REVIEW

File: src/renderer/components/Tables/VideoSelectionTable.tsx [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+        id: 'Details',
+        size: 50,
+        cell: populateDetailsCell,
+      },
+    ],
+    []
+  );
+
+  const { category, playingVideo } = appState;
+  let columns = raidColumns;
```
Comment (aza547) [inline src/renderer/components/Tables/VideoSelectionTable.tsx:342]: switch for this

File: src/renderer/components/Tables/VideoSelectionTable.tsx [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+    stateManager,
+    persistentProgress,
+  } = props;
+
+  const [expanded, setExpanded] = useState<ExpandedState>({});
+
+  /**
+   * Reset expanded on changing category.
+   */
+  useEffect(() => {
```
Comment (aza547) [inline src/renderer/components/Tables/VideoSelectionTable.tsx:90]: seems like this would be better set in the select category handler, higher in the stack

**Comment by aza547** [comment]
Probably done here and this is fine to go into main. 
@StephixOne flagging in-case if you fancy a look, welcome any comments. Np if not.

**Comment by aza547** [comment]
Ah I forgot I made some inline comments. Will resolve those tomorrow I guess.

**Comment by StephixOne** [comment]
Happy to have a better look at the weekend, busy week.

**Comment by aza547** [comment]
> Happy to have a better look at the weekend, busy week.

Thanks! Might be an idea to just try it out also - builds on this branch should be good.

**Comment by aza547** [comment]
Some weird overflow maybe on the expand button?

{88BA2C03-2A5E-41A0-9812-668A991DC080}

File: src/renderer/components/Tables/VideoSelectionTable.tsx (hunk 2 above)
Comment (aza547) [reply src/renderer/components/Tables/VideoSelectionTable.tsx:93]: I left a comment. Not changing for now.

**Comment by aza547** [comment]
Think I'm actually done now.

image

Not sure I'm totally happy with the contents of the expanded box but don't want to let perfect be the enemy of good. I think this is a big improvement.

**Comment by aza547** [comment]
Going for the merge. Let me know if you get a chance to look and have any retrospective comments @StephixOne  :)

**Pull request merged**